### PR TITLE
Add Variant Description and pageview totals to Experiment Results res…

### PR DESCRIPTION
**Pull request by freddyucv**
We need to include into the Response of the Experiment's Results End Point:

- The Variant's Description.
- The total of page view by Variant

### Some Refactoring

- Move the filters from String to Object in the CubeJsQueryBuilder filter method

https://github.com/dotCMS/core/pull/24625/files#diff-c50354f1a06c2e8525d71ddbb42ebb62b376ba7a22d238ff714e76584d3c3674R279

https://github.com/dotCMS/core/pull/24625/files#diff-b9cd675fa490c8ec6e6ea7c221ee177d2481c6d60070f747e70316e045201a2fR67

- Use the GoalBuilder in  a best self explain way.

https://github.com/dotCMS/core/pull/24625/files#diff-2e2f84d7d5a179b103ccc74167aa1634eb65d47eb9debf53e265966fe8bd0168R37

https://github.com/dotCMS/core/pull/24625/files#diff-640f35cbf0cfbfe8f83a40233ceb94d934228a48ac2e269bc3b1f138b04e6363R93

https://github.com/dotCMS/core/pull/24625/files#diff-10fc5a5c8d8484aa3f69421a60d72d73aebe0a65413ce9023843ddbf7f9a38c7R43

- Use the VariantResultBuilder in  a best self explain way.

https://github.com/dotCMS/core/pull/24625/files#diff-c344390af71188fae79256148827c3cff25c4b23af6eec4b60e77f82cce4dee8R24

## Include the PageViews into the Response

- Calculate PageViews by Variant

https://github.com/dotCMS/core/pull/24625/files#diff-1fdf3602f53e848245929cb77e96e83012c452faf87ae01d3e699826cfd22f2eR125

- Set the Page views total into the Builder

https://github.com/dotCMS/core/pull/24625/files#diff-1fdf3602f53e848245929cb77e96e83012c452faf87ae01d3e699826cfd22f2eR88

- Include the PageView total into the Response

https://github.com/dotCMS/core/pull/24625/files#diff-dceceabfd37a1dd630c7fab22a6ce17664fd6d5a39dd30366e22f4aa11e3643cR58

## Improve the MockHttpServer to allow more than one Context to the same URI

This mean that now it can response with two different mock response in the same URL according to the request paramater.

https://github.com/dotCMS/core/pull/24625/files#diff-bb8e6d4042484875cf32b80f746a82828bf3c93ebaa404c732329defc4f9b003R60

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/cube/filters/SimpleFilter.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public Operator getOperator() {
     }
 
     public String[] getValues() {
-        return values;
+        return Arrays.stream(values).map(Object::toString).toArray(String[]::new);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/cube/filters/SimpleFilter.java:67]: be defensive, check if values is not before transformation

File: dotCMS/src/main/java/com/dotcms/experiments/business/result/ExperimentAnalyzerUtil.java [hunk 2]
```diff
@@ -28,6 +40,8 @@ public enum ExperimentAnalyzerUtil {
 
     INSTANCE;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/experiments/business/result/ExperimentAnalyzerUtil.java:42]: This looks such as WebAPI

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java [hunk 3]
```diff
@@ -501,6 +539,131 @@ public void pageReachDirectReferer() throws DotDataException, DotSecurityExcepti
...
+            final AnalyticsHelper mockAnalyticsHelper = mockAnalyticsHelper();
+
+            ExperimentAnalyzerUtil.setAnalyticsHelper(mockAnalyticsHelper);
+
+            final ExperimentsAPIImpl experimentsAPIImpl = new ExperimentsAPIImpl(mockAnalyticsHelper);
+
+            final ExperimentResults experimentResults = experimentsAPIImpl.getResults(experiment);
+
+            mockhttpServer.validate();
+
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java:628]: Looks like the test assert is missing

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/http/server/mock/MockHttpServer.java [hunk 4]
```diff
@@ -75,6 +85,53 @@ public void start() {
         }
     }
 
+    private static String getErrorMessage(List<MockHttpServerContext> contexts,
+            RequestContext requestContext) {
+        return "New Request:" + contexts.stream()
+                    .flatMap(context -> context.getRequestConditions().stream()
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/http/server/mock/MockHttpServer.java:91]: If intend to use `flatMap()` then I think the code you will need it's actually:
```
contexts.stream()
                    .flatMap(context -> context.getRequestConditions().stream())
                    .map(reqCondition -> requestCondition.getMessage(requestContext)
```

File: dotCMS/src/main/java/com/dotcms/experiments/business/result/ExperimentAnalyzerUtil.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/experiments/business/result/ExperimentAnalyzerUtil.java:42]: Not really, this really analyze the Experiments Events to calculate the results, so really it is connect to CubeJs. nothing to do with a request so I think is not really a WebApI class

Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/experiments/business/result/ExperimentAnalyzerUtil.java:42]: Not really, this really analyze the Experiments Events to calculate the results, so really it is connect to CubeJs. nothing to do with a request so I think is not really a WebApI class

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java:678]: done https://github.com/dotCMS/core/pull/24625/commits/989229988f13072ff026d0e585bd2ddb61304cc3#diff-25dea0f580007d55e9704544452fcf0833029a2294fb814f4dddfb1bb2e4ba9bR681

File: dotCMS/src/main/java/com/dotcms/cube/filters/SimpleFilter.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/cube/filters/SimpleFilter.java:71]: done https://github.com/dotCMS/core/pull/24625/commits/989229988f13072ff026d0e585bd2ddb61304cc3#diff-b9cd675fa490c8ec6e6ea7c221ee177d2481c6d60070f747e70316e045201a2fR67

File: dotCMS/src/main/java/com/dotcms/http/server/mock/MockHttpServer.java (hunk 4 above)
Comment (freddyucv) [reply dotCMS/src/main/java/com/dotcms/http/server/mock/MockHttpServer.java]: done https://github.com/dotCMS/core/pull/24625/commits/989229988f13072ff026d0e585bd2ddb61304cc3#diff-bb8e6d4042484875cf32b80f746a82828bf3c93ebaa404c732329defc4f9b003R91

**Pull request merged**